README.md

```diff
@@ -137,17 +155,15 @@ Instantiate a new `Storj` object used to communicate with the storj network. The
 
 If you need to use authentication in your application, we strongly suggest you use the `key` method as it provides a higher level of security.
 
-Both `basicAuth` and `key` are optional, but you may only provide one or the other. If you use `basicAuth`, the library will attempt to create a public/private key pair for you and register it to your account. If you provide a `key`, this key will be used to authenticate every request moving forward.
+Both `basicAuth` and `key` are optional, but you may only provide one or the other. If you use `basicAuth`, the library will assume that you already have registered a public key with the bridge you are authenticating with. To create a public/private key pair for you and register it to your account you can use the `getKeypair` and `registerKeypair` API. If you provide a `key`, this key will be used to authenticate every request moving forward.
```

Comment:
Yeah we should change the documentation. Here is how I see it working. 

```
var client = {
  bridge: bridge,
  // either supply basic auth or a string private key that will be autogenerated into a keypair
```

Currently we are not handling user creation or sign-up with bridge since it relies on smtp. We will just assume that there is a `basicAuth` strategy for everyone and this strategy will be required only when calling `storj.registerKey()`. 

Now we can also assume from this point a user has never paired with the bridge thus they have a `basicAuth` strategy assigned to `storj`.

They will then `storj.registerKeypair()` This will place a newly generated `keypair` on the `storj` object and register it with the bridge for further auth. Otherwise if you have already registered your public key then all auth will use the user private key supplied to the client constructor.

So if you are calling `storj.createFile()` it will use whatever strat is supplied on the `client` object and fail from server if none is. The first thing `createFile` does now is ping the bridge for a token. If this token has a private key in it, then we know we are in a public bucket and we will override the `File` encryption key (not the `Storj` user private key) with the buckets key and use that, otherwise it will need to find the private key in our keyring based on the bucket we uploading to. This has not been implemented yet.